kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -256,6 +257,21 @@ ExpressionTransformResult visitCoalesce(ScalarExpression coalesce) {
           children.get(0).outputType);
     }
 
+    @Override
+    ExpressionTransformResult visitAdd(ScalarExpression add) {
+      List<ExpressionTransformResult> children =
+          add.getChildren().stream().map(this::visit).collect(Collectors.toList());
+      if (children.size() != 2) {
+        throw unsupportedExpressionException(
+            add, "ADD requires exactly two arguments: left and right operands");
+      }
+
+      return new ExpressionTransformResult(
+          new ScalarExpression(
+              "ADD", Arrays.asList(children.get(0).expression, children.get(1).expression)),
+          children.get(0).outputType);
```

Comment:
Similar to e.g. coalesce above, we should probably start by requiring both arguments to have the exact same type.
Otherwise, we need to define type coercion rules to infer the resulting type of ADD - e.g. int + decimal = ? - and that's a big undertaking

Ideally we'd also check that the type is one of the supported numeric type by the underlying implementation you provide, e.g. String + String shouldn't work